pandas/tests/plotting/test_datetimelike.py

```diff
@@ -291,9 +292,16 @@ def test_irreg_hf(self):
         _, ax = self.plt.subplots()
         df2 = df.copy()
         df2.index = df.index.astype(object)
-        df2.plot(ax=ax)
-        diffs = Series(ax.get_lines()[0].get_xydata()[:, 0]).diff()
-        assert (np.fabs(diffs[1:] - sec) < 1e-8).all()
+        with warnings.catch_warnings(record=True) as w:
```

Comment:
I would rather use `assert_produces_warning` than custom code (and just disable the stack level check).  pls also coment that the explict cast to object is causing.

Reply:
Done